src/Domain/DonationRepository.php

```diff
@@ -186,45 +186,32 @@ public function allocateMatchFunds(Donation $donation): string
         // is most often empty (for new donations) so this will frequently be 0.00.
         $amountMatchedAtStart = $donation->getFundingWithdrawalTotal();
 
-        /** @var FundingWithdrawal[] $newWithdrawals */
-        $newWithdrawals = [];
-
-        while (!$allocationDone && $allocationTries < $this->maxLockTries) {
-            try {
-                $likelyAvailableFunds = $this->getEntityManager()
-                    ->getRepository(CampaignFunding::class)
-                    ->getAvailableFundings($donation->getCampaign());
+        try {
+            $likelyAvailableFunds = $this->getEntityManager()
+                ->getRepository(CampaignFunding::class)
+                ->getAvailableFundings($donation->getCampaign());
```

Comment:
The old version of the code looks like it always executes the loop body exactly once.